src/weapon/w_026.c

```diff
@@ -50,24 +50,209 @@ void func_BA000_8017B37C(void) {
+                        prim->next = nextPrim;
+                        prim = prim->next;
+                        // Note: In this case, we do i++ but not prim =
+                        // prim->next
+                    } else {
+                        prim->drawMode |= DRAW_HIDE;
+                        prim = prim->next;
+                    }
+                } else {
+                    prim = prim->next;
+                }
+            }
+            return;
+        } else {
+            self->ext.shamanshieldstar.unk7E +=
+                self->ext.shamanshieldstar.unk96;
+            if (self->ext.shamanshieldstar.unk80 == 0) {
+                self->ext.shamanshieldstar.unk9C -= 6;
+                if (self->ext.shamanshieldstar.unk9C <= 0) {
+                    self->ext.shamanshieldstar.unk9C = 0;
+                }
+                self->ext.shamanshieldstar.unk9E -= 6;
+                if (self->ext.shamanshieldstar.unk9E <= 0) {
+                    self->ext.shamanshieldstar.unk9E = 0;
+                }
+            } else {
+                self->ext.shamanshieldstar.unk80--;
+            }
+            if (self->ext.shamanshieldstar.parent->step == 8) {
+                self->step++;
+                self->ext.shamanshieldstar.unk80 = rand() & 0xF;
+            }
+        }
+        break;
+    case 2:
+        self->ext.shamanshieldstar.unk7E += self->ext.shamanshieldstar.unk96;
+        if (self->ext.shamanshieldstar.unk80 == 0) {
+            self->ext.shamanshieldstar.unk9C += 6;
+            self->ext.shamanshieldstar.unk9E -= 6;
+            if (self->ext.shamanshieldstar.unk9E < -0x80) {
+                DestroyEntity(self);
+                return;
+            }
+            break;
+        }
+        self->ext.shamanshieldstar.unk80--;
+    }
+
+    vectors[0].vx += self->ext.shamanshieldstar.unk9C;
+    vectors[1].vx += self->ext.shamanshieldstar.unk9C;
+    vectors[2].vx += self->ext.shamanshieldstar.unk9C;
+    vectors[3].vx += self->ext.shamanshieldstar.unk9C;
+
+    vectors[0].vy += self->ext.shamanshieldstar.unk9E;
+    vectors[1].vy += self->ext.shamanshieldstar.unk9E;
+    vectors[2].vy += self->ext.shamanshieldstar.unk9E;
+    vectors[3].vy += self->ext.shamanshieldstar.unk9E;
+
+    rotation_axis.vx = self->ext.shamanshieldstar.unk7C;
+    rotation_axis.vy = self->ext.shamanshieldstar.unk7E;
+    rotation_axis.vz = self->ext.shamanshieldstar.unk82;
+
+    transfer_vector.vz = self->ext.shamanshieldstar.unk9A;
+    transfer_vector.vx = 0;
+    transfer_vector.vy = 0x10;
+
+    SetGeomScreen(0x280);
+    SetGeomOffset(self->posX.i.hi, self->posY.i.hi);
+    RotMatrix(&rotation_axis, &m);
+    TransMatrix(&m, &transfer_vector);
+    SetRotMatrix(&m);
+    SetTransMatrix(&m);
+    prim = &g_PrimBuf[self->primIndex];
+
+    rotTransResult = RotTransPers4(
+        vecs[0][0], vecs[0][1], vecs[0][2], vecs[0][3], (long*)&prim->x0,
+        (long*)&prim->x1, (long*)&prim->x3, (long*)&prim->x2, &spA0, &spA4);
```

Comment:
spA0/A4 could be renamed but they are unused anyway
<img width="678" alt="image" src="https://github.com/Xeeynamo/sotn-decomp/assets/122322823/12e365b7-2deb-44d5-851c-e72a23b85ed6">

Reply:
Yeah, I figured if they are unused there's not much point in naming them